bundles/org.eclipse.equinox.slf4j/src/org/eclipse/equinox/slf4j/EquinoxLoggerFactory.java

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.equinox.slf4j;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.eclipse.core.internal.runtime.PlatformLogWriter;
+import org.eclipse.equinox.log.ExtendedLogService;
+import org.eclipse.equinox.log.Logger;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.ILoggerFactory;
+
+/**
+ * An implementation of the SLF4J {@link ILoggerFactory} that is using the
+ * equinox platform log like done with the ILog interface
+ */
+public class EquinoxLoggerFactory implements org.slf4j.ILoggerFactory {
+
+	private static final StackWalker STACK_WALKER = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+	private static final Bundle FACTORY_BUNDLE = FrameworkUtil.getBundle(EquinoxLoggerFactory.class);
+	private static final Bundle SLF4J_BUNDLE = FrameworkUtil.getBundle(org.slf4j.ILoggerFactory.class);
+	private static final ConcurrentMap<Bundle, EquinoxLogger> LOGGER_MAP = new ConcurrentHashMap<>();
+	private ExtendedLogService logService;
```

Comment:
Can you get by with the LogService here?

Reply:
Not really because `ExtendedLogService` has a special method with name+bundle that is also used internally in Platform, and I wanted to be en-par with that.

Reply:
Pretty sure you can get by with `org.osgi.service.log.LoggerFactory.getLogger(Bundle, String, Class<L>)` which `LogService extends LoggerFactory`

Reply:
I first thought of that as well but it has additional check on the bundle and bundlestate :-\

Also we here explicitly reference the framework logger so I don't see much of an benefit using "standard" API.

Reply:
> I first thought of that as well but it has additional check on the bundle and bundlestate :-\
> 
> Also we here explicitly reference the framework logger so I don't see much of an benefit using "standard" API.

Not sure why you want this slf4j stuff to work with a `null` bundle or one that is not resolved.

Reply:
Main point is I want it to make sure it is the Equinox Log service, so I can use that other method probably but will it help something?!?

Reply:
I now use `LogService` api as much as possible